src/vm/callingconvention.h

```diff
@@ -1701,6 +1701,17 @@ inline BOOL HasRetBuffArg(MetaSig * pSig)
     return argit.HasRetBuffArg();
 }
 
+#ifdef UNIX_X86_ABI
+inline BOOL HasRetBuffArgUnmanagedFixup(MetaSig * pSig)
+{
+    WRAPPER_NO_CONTRACT;
+    CorElementType type = pSig->GetReturnTypeNormalized();
+
+    // For UNIX_X86_ABI and unmanaged function, we always need RetBuf if the return type is VALUETYPE
+    return type == ELEMENT_TYPE_VALUETYPE;
```

Comment:
Since `void ArgIteratorTemplate<ARGITERATOR_BASE>::ComputeReturnFlags()` uses `GetReturnType()` as well. 

https://github.com/dotnet/coreclr/blob/19e1cd299da301d1b2cb18ae8479595dbfc7224c/src/vm/callingconvention.h#L1329